pyrokinetics/pyro.py

```diff
@@ -1093,6 +1093,27 @@ def local_geometry_type(self) -> Union[str, None]:
         else:
             raise TypeError("Pyro._local_geometry is set to an unknown geometry type")
 
+    def switch_local_geometry(self, local_geometry=None, **kwargs):
+        """
+        Switches LocalGeometry type
+        Returns
+        -------
+
+        """
+
+        # Check if already loaded and if show then switch geometries
+        if not isinstance(self.local_geometry, LocalGeometry):
+            raise ValueError("Please load local geometry before switching")
+
+        if local_geometry in self.supported_local_geometries:
```

Comment:
If it's not supported, we should raise a `ValueError` instead, e.g.:

```suggestion
        if local_geometry not in self.supported_local_geometries:
            raise ValueError(f"Unsupported local geometry type. Got '{local_geometry}', expected one of: {self.supported_local_geometries.keys()}")
```

Then rest of method can be unindented one step too :)